pilot/pkg/serviceregistry/kube/controller/ambient/ambientindex.go

```diff
@@ -156,14 +158,15 @@ func New(options Options) Index {
 		XDSUpdater:                  options.XDSUpdater,
 		Debugger:                    options.Debugger,
 		Flags:                       options.Flags,
+		revision:                    options.Revision,
 		clientBuilder:               options.ClientBuilder,
 		stop:                        make(chan struct{}),
 		cs:                          multicluster.NewClustersStore(),
 		remoteClientConfigOverrides: options.RemoteClientConfigOverrides,
 	}
 
 	filter := kclient.Filter{
-		ObjectFilter: options.Client.ObjectFilter(),
+		ObjectFilter: kubetypes.ComposeFilters(options.Client.ObjectFilter(), a.inRevision),
```

Comment:
This should only be on configuration objects to maintain consistency. So not Services seems to be the only one violating this currently. Maybe renaming `filter` to make that clear would help avoid future mis-use as well